Pjiwm/kaizominecraft
# Generating custom worlds for kaizo

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

**Please provide enough information so that others can review your pull request:**

Apart from the normal Minecraft dimensions there should be a Kaizo variant of them.
The worlds should be linked together so when a player goes in a portal they still stay in the Kaizo variant.
All kaizo worlds need custom terrain which is created with minecraft own json system for creating custom worlds.

**Explain the **details** for making this change. What existing problem does the pull request solve?**

Upon starting up the server there should be a check if the worlds kaizo, kaizo_nether and kaizo_end exist.
If not these world should be created with custom generation parameters. 
These custom generation parameters need to be made with json that tells the plugin what to edit about the world.
The terrains should be made more difficult to traverse and give an over all more difficult feeling to them.
The following list of components should be added to implement this feature:

- [x] A world creation class that generates new custom worlds.
- [x] ~~3 json files/json strings that describe the custom terrain of each Minecraft dimension.~~
- [x] a way for a player to enter the world. This can be done via events or commands.
- [x] a way for the player to return back to the normal worlds. (might scrap or make admin only)
- [x] a listener that checks portal events in kaizo worlds so that the player stays in the kaizo worlds.
- [x] spawnpoint of the player should be changed upon entering kaizo dimension.
- [x] store seperate inventory and player data in kaizo world
- [x] change custom mob events that now only occur in the default minecraft world to kaizo variant.
- [x] prevent buffed mobs to spawn in normal world.

**Test plan (required):**
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request has visual changes.
1. remove kaizo world folders until the results are correct
2. join minecraft server and make sure player is in a kaizo world
3. check if features work within kaizo world

**Closing issues**

closes #11 
Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
